examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1344 @@
+  template <int dim>
+  class NormalGradientVolumeStabilization
+  {
+  public:
+    NormalGradientVolumeStabilization(VectorType &solution,
+                                      VectorType &level_set)
+      : solution(solution)
+      , level_set(level_set)
+      , stabilization_parameter(1.0)
+      , stabilization_exponent(-1.0)
+    {}
+
+    // Since  the stabilization form contributes to the error computation,
+    //  we pass the discrete solution and the discrete level-set to the class
+    //  which refer to different dof-handlers.
+    VectorType &solution;
+    VectorType &level_set;
+    // Clearly, the next method indicates that the assembly and the evaluation
+    // of the form does require a bulk cell quadrature. This methodology may be
+    // utilized in the MeshWorker.
+    bool needs_cell_worker()
+    {
+      return true;
+    }
+    // We define the stabilization form here assuming that ScratchData and
+    // CopyData arguments are initialized properly. The local contribution of
+    // the stabilization on this cell is defined.
+    template <class Iterator>
+    void cell_worker_assemble(const Iterator            &cell,
+                              TraceFEM_ScratchData<dim> &scratch_data,
+                              TraceFEM_CopyData<dim>    &copy_data)
+    {
+      const FEValues<dim> &fe_values = scratch_data.fe_values;
+      const FEValues<dim> &level_set_fe_values =
+        scratch_data.level_set_fe_values;
+      // These points and weights are for the bulk cell quadrature  that is
+      // suitable for the discrete level-set.
+      const std::vector<double> &cellJxW = fe_values.get_JxW_values();
+
+      // Similarly, the gradients of the discrete level-set are computed
+      //  in the bulk cell quadrature, which, upon normalization, give the
+      //  discrete
+      // normal vector in a bulk cell.
+      std::vector<Tensor<1, dim>> grad_level_set(
+        level_set_fe_values.get_quadrature().size());
+      level_set_fe_values.get_function_gradients(level_set, grad_level_set);
+
+      double factor =
+        stabilization_parameter *
+        std::pow(cell->minimum_vertex_distance(), stabilization_exponent);
+      for (const unsigned int q : fe_values.quadrature_point_indices())
+        {
+          const Tensor<1, dim> &normal =
+            (1.0 / grad_level_set[q].norm()) * grad_level_set[q];
+          // Finally, the CopyData matrix computes the cell contribution
+          for (const unsigned int i : fe_values.dof_indices())
+            for (const unsigned int j : fe_values.dof_indices())
+              copy_data.cell_matrix(i, j) +=
+                factor * (normal * fe_values.shape_grad(i, q)) *
+                (normal * fe_values.shape_grad(j, q)) * cellJxW[q];
+        }
+    }
+
+    // Similarly, we evaluate the stabilization form on this cell.
+    template <class Iterator>
+    void cell_worker_evaluate(const Iterator            &cell,
```

Comment:
The name of the function is a bit unfortunate. It does not indicate that you are evaluating the error.